src/pages/welcome.tsx

```diff
@@ -471,6 +483,353 @@ const DataCheckSection = ({
+                      { value: "vaxzevria", label: "Vaxzevria (AstraZeneca)" },
+                      { value: "spikevax", label: "Spikevax (Moderna)" },
+                      { value: "jcovden", label: "Jcovden (J&J)" },
+                      { value: "covilo", label: "Covilo (Sinopharm)" },
+                      {
+                        value: "covovax",
+                        label: "COVOVAX (Novavax formulation)",
+                      },
+                    ]}
+                    onChange={(e: string) => setForm({ ...form, provider: e })}
+                    defaultValue={"pfizer"}
+                  />
+                </div>
+                <div className="layout-grid-cols-2 !gap-y-0">
+                  <KeyboardInput
+                    name="vaccine-lot"
+                    type="text"
+                    label={t("vaccine.lot.label", { ns: "covid" })}
+                    onChange={(e) => setForm({ ...form, lotno: e })}
+                    attr={{ disabled }}
+                  />
+                  <KeyboardInput
+                    name="vaccine-administering-center"
+                    type="text"
+                    label={t("vaccine.administeringCenter.label", {
+                      ns: "covid",
+                    })}
+                    onChange={(e) =>
+                      setForm({ ...form, administeringCenter: e })
+                    }
+                    attr={{ disabled }}
+                  />
+                </div>
+              </section>
+              <CardActions>
+                <Button
+                  label={t("welcome.vaccineData.action.add")}
+                  type="filled"
+                  onClick={async () => {
+                    const { data: personid } = await getPersonIDFromUser(
+                      supabase,
+                      user!
+                    );
+                    toggleLoading();
+                    await addVaccineRecord(
+                      supabase,
+                      {
+                        id: 0,
+                        doseNo: 0,
+                        lotNo: form.lotno,
+                        vaccineDate: new Date(form.date),
+                        administeredBy: form.administeringCenter,
+                        vaccineName: form.provider,
+                      },
+                      personid!
+                    );
+                    const newVaccineRecords = await getVaccineRecordbyPersonId(
+                      supabase,
+                      personid!
+                    );
+                    toggleLoading();
+                    setVaccineData(newVaccineRecords);
+                  }}
+                  disabled={disabled || !validateForm() || loading}
+                />
+              </CardActions>
+            </Card>
+          </div>
+
+          <div className="col-span-3 md:col-start-5">
+            {vaccineData.length > 0 &&
```

Comment:
So nothing appears when there are no records? I think this Section would feel more complete if there’s a box on the right saying there are no records. Too much white space in a design is not a good thing.